src/interfaces/index.ts

```diff
@@ -0,0 +1 @@
+export * from './permissions';
```

Comment:
I'm a fan of barrel exporting for the following reasons:
- less import code at the top of files and easier to read the location of those imports. 
- easier mocking/dependency injection
- allows other modules to leverage the barreled imports as a dynamic map

Reply:
Barrel export for external consumption makes total sense to me. For internal consumption, not as much, because:

1. currently I am not seeing a clear pattern when `index.ts` should or should not be created
2. there are more than "just" barrel exports in some `index.ts` (exporting a map), which is seems to be bending the intent of barrel exports to my understanding
3. i generally do not look at import statements, so value-add of less lines is "meh"

Don't understand point 2 & 3, but again you clearly are making deliberate points that I am not yet grasping so won't block the PR on this. Will leave open for discussion.